File: sdks/php/src/Types/ResolvedTree.php
<?php

namespace Autonoma\Sdk\Types;

class ResolvedTree
{
    /** @var CreateOp[] */
    public array $ops = [];

    /** @var array<string, string> alias -> temp id */
    public array $aliases = [];

    /** @var array<string, string> alias -> model name */
    public array $aliasOwnerModel = [];

    /** @var array<string, string[]> alias -> list of dependency aliases */
    public array $aliasDependencies = [];
}

File: sdks/java/autonoma-sdk/src/main/java/ai/autonoma/sdk/types/ResolvedTree.java
package ai.autonoma.sdk.types;

import java.util.List;
import java.util.Map;

public record ResolvedTree(
    List<CreateOp> ops,
    Map<String, String> aliases,
    Map<String, String> aliasOwnerModel,
    Map<String, List<String>> aliasDependencies
) {}
